pinot-transport/src/main/java/com/linkedin/pinot/requestHandler/BrokerRequestHandler.java

```diff
@@ -90,23 +93,47 @@
   private final BrokerRequestOptimizer _optimizer;
   private final int _queryResponseLimit;
   private AtomicLong _requestIdGenerator;
+  private Configuration _config;
+  private final String _brokerId;
 
   //TODO: Currently only using RoundRobin selection. But, this can be allowed to be configured.
   private RoundRobinReplicaSelection _replicaSelection;
 
+  private static final int DEFAULT_BROKER_QUERY_RESPONSE_LIMIT = Integer.MAX_VALUE;
+  private static final String BROKER_QUERY_RESPONSE_LIMIT_CONFIG = "pinot.broker.query.response.limit";
+  public  static final long DEFAULT_BROKER_TIME_OUT_MS = 10 * 1000L;
+  private static final String BROKER_TIME_OUT_CONFIG = "pinot.broker.timeoutMs";
+  public static final String BROKER_ID_CONFIG_KEY = "pinot.broker.id";
+
   public BrokerRequestHandler(RoutingTable table, TimeBoundaryService timeBoundaryService,
       ScatterGather scatterGatherer, ReduceServiceRegistry reduceServiceRegistry, BrokerMetrics brokerMetrics,
-      long brokerTimeOutMs, int queryResponseLimit) {
+      Configuration config) {
     _routingTable = table;
     _timeBoundaryService = timeBoundaryService;
     _reduceServiceRegistry = reduceServiceRegistry;
     _scatterGatherer = scatterGatherer;
     _replicaSelection = new RoundRobinReplicaSelection();
     _brokerMetrics = brokerMetrics;
-    _brokerTimeOutMs = brokerTimeOutMs;
+    _config = config;
     _optimizer = new BrokerRequestOptimizer();
     _requestIdGenerator = new AtomicLong(0);
-    _queryResponseLimit = queryResponseLimit;
+    _queryResponseLimit = _config.getInt(BROKER_QUERY_RESPONSE_LIMIT_CONFIG, DEFAULT_BROKER_QUERY_RESPONSE_LIMIT);
+    _brokerTimeOutMs = _config.getLong(BROKER_TIME_OUT_CONFIG, DEFAULT_BROKER_TIME_OUT_MS);
+    _brokerId = readBrokerIdFromConfig();
+    LOGGER.info("Broker response limit is: " + _queryResponseLimit);
+    LOGGER.info("Broker timeout is - " + _brokerTimeOutMs + " ms");
+    LOGGER.info("Broker id: " + _brokerId);
+  }
+
+  private String readBrokerIdFromConfig() {
+    String brokerIdValue = null;
+    try {
+      brokerIdValue = _config.getString(BROKER_ID_CONFIG_KEY, InetAddress.getLocalHost().getHostName());
```

Comment:
Is there any impact on low latency usecases (say with latency in sub 10ms) due to this new code, specifically the getHostName() call?